### PR TITLE
[ADD] stock_sale_crm

### PR DESCRIPTION
**Pull request by veloutin**
This module expands the behavior of sale_crm when creating invoices from sales order to delivery orders that have an associated sales order: the salesteam is used in the invoice if set.

### CODE REVIEW

File: stock_sale_crm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from . import stock
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (bwrsandman) [inline stock_sale_crm/__init__.py]: Please don't include vim lines

File: stock_sale_crm/stock.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp.osv import orm
+
+
+class stock_picking(orm.Model):
+    _inherit = "stock.picking"
```
Comment (bwrsandman) [inline stock_sale_crm/stock.py:27]: Please name the file after the model `stock_picking.py`

**Comment by bwrsandman** [comment]
:+1:

**Comment by max3903** [comment]
Please use the following information:

Module name: account_invoice_default_salesteam
Description: """
This module set the Sales Team (section_id field) in the 2 following scenarios :
- when creating an invoice, set on partner_id.onchange() from the partner's Sales Team
- when invoicing from sales_order, set from the Sales Order's Sales Team.
  """

File: stock_sale_crm/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+{
+    'name': 'Invoice Salesteam on Delivery Order',
```
Comment (max3903) [inline stock_sale_crm/__openerp__.py:24]: 'name': 'Default Sales Team on Invoice',

File: stock_sale_crm/stock.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp.osv import orm
+
+
+class stock_picking(orm.Model):
```
Comment (pedrobaeza) [inline stock_sale_crm/stock.py]: Use CapsWord convention: `class StockPicking...`

**Comment by max3903** [comment]
Instead of setting the sales team when preparing the invoice from the delivery order, wouldn't it be better to set the sales team using defaults when an invoice is created, wherever you are creating it from and use the source document to set the value ?

File: stock_sale_crm/stock.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+#
+###############################################################################
+
+from openerp.osv import orm
+
+
+class StockPicking(orm.Model):
+    _inherit = "stock.picking"
+
+    def _prepare_invoice(self, cr, uid, picking, partner, inv_type, journal_id, context=None):
```
Comment (pedrobaeza) [inline stock_sale_crm/stock.py]: This line doesn't respect PEP8 80 cols limit

File: stock_sale_crm/stock.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        """ Builds the dict containing the values for the invoice
+            @param picking: picking object
+            @param partner: object of the partner to invoice
+            @param inv_type: type of the invoice ('out_invoice', 'in_invoice', ...)
```
Comment (pedrobaeza) [inline stock_sale_crm/stock.py]: This line doesn't respect PEP8 80 cols limit

**Comment by eLBati** [comment]
Thanks @veloutin any chances to have tests?

**Comment by veloutin** [comment]
Replaced by #33

**Pull request closed without merging**